apis/management.cattle.io/v3/machine_types.go

```diff
@@ -42,10 +42,11 @@ type NodeTemplateCondition struct {
 }
 
 type NodeTemplateSpec struct {
-	DisplayName         string `json:"displayName"`
-	Description         string `json:"description"`
-	Driver              string `json:"driver" norman:"nocreate,noupdate"`
-	CloudCredentialName string `json:"cloudCredentialName" norman:"type=reference[cloudCredential]"`
+	DisplayName         string     `json:"displayName"`
+	Description         string     `json:"description"`
+	Driver              string     `json:"driver" norman:"nocreate,noupdate"`
+	CloudCredentialName string     `json:"cloudCredentialName" norman:"type=reference[cloudCredential]"`
+	NodeTaints          []v1.Taint `json:"nodeTaints,omitempty"`
```

Comment:
is there a reason for adding taints, but not nodeLabels/nodeAnnotations, on the node pool? Usually they all come together at other places.

Reply:
This is the requirement from @cjellick and @vincent99. They want to set node taints from node pool and node template. And set those taints for newly created node from node pool.

Reply:
The node labels from the template are stored in `NodeTemplate.ObjectMeta.Labels` so the node label field don't need to defined in the `NodeTemplateSpec` struct.

Reply:
Yea, my question was more about "can we add similar support for labels/annotations on the pool". But if they get inherited from the object meta, and already working, perfect.